dotnet/Razorvine.Pyrolite/Pyrolite/Pyrolite.csproj

```diff
@@ -3,7 +3,8 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <AssemblyName>Razorvine.Pyrolite</AssemblyName>
     <RootNamespace>Razorvine</RootNamespace>
-    <LangVersion>6</LangVersion>
+    <LangVersion>7.3</LangVersion>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```

Comment:
is this still needed? that unsafe method rawStringFromBytes() was already corrected i believe

Reply:
> is this still needed?

It is.  There were four places `unsafe` is used in my changes; I'll revert the unnecessary one that @eerhardt spotted, as that was just a copy and paste mistake on my part, but the others are still intentional.

Does this concern you?  I obviously don't want to push for anything that makes you uncomfortable.

Reply:
It doesn't really, seems like you know what you're doing -- it's just that I never have had to use unsafe in my own code so far.

Reply:
Yeah, `unsafe` isn't strictly needed here for functionality... obviously it was functional before my suggested changes (with a goal of the changes to be not affecting functionality at all).  It's being used to help with perf, and that's generally where `unsafe` comes into play.

The specific places I'm using it:
- BinaryPrimitives doesn't currently expose APIs for working with floating-point numbers.  So whereas for an Int32 I can make a call like BinaryPrimitives.ReadInt32BigEndian, I can't do that for a float.  For double, I can get around this by using BinaryPrimitives.ReadInt64BigEndian and then reinterpreting that as a double with BitConverter.Int64BitsToDouble, but a corresponding conversion method doesn't exist in .NET Standard 2.0 for Int32->Single (one was added for .NET Core 3.0 and .NET Standard 2.1, though).  With unsafe, code, though, we can easily achieve that with `*(float*)&int32Value`, which as it happens is exactly the implementation of Int32BitsToSingle in .NET Core: https://github.com/dotnet/coreclr/blob/87ef27bf081432457e6f1fd635a765b147406a96/src/System.Private.CoreLib/shared/System/BitConverter.cs#L464-L468
- The `rawStringFromBytes` function could be written without unsafe code by first creating a char[], copying the data into that, and then creating the string from that char[]… but that involves an allocation, which is what I was trying to avoid.  With .NET Core 2.1 and .NET Standard 2.1, it could also be done using the new `string.Create` method, which is actually what I would prefer to do, but I couldn't here because it's not available with .NET Standard 2.0.  So I instead employ this trick with `unsafe`.

I mentioned three earlier... in reviewing them to outline the above, I noticed that one of the places I'm using it isn't actually necessary, so I'll revert that one, too.